File: 240819/연속되는 수 3/continuous-number3.py
n = int(input())

array = []
for _ in range(n):
    x = int(input())
    array.append(x)

answer = 0
cnt = 1
for i in range(1, len(array)):
    if array[i - 1] > 0 and array[i] > 0:
        cnt += 1
    elif array[i - 1] < 0 and array[i] < 0:
        cnt += 1
    else:
        cnt = 1

    answer = max(answer, cnt)

print(answer)

File: 241026/연속되는 수 3/continuous-number3.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int[] arr = new int[n];

        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
    
        int cnt = 0;
        int maxVal = Integer.MIN_VALUE;
        for (int i = 0; i < n; i++) {
            if (i == 0 || arr[i] * arr[i - 1] > 0) {
                cnt += 1;
            } else {
                cnt = 1;
            }

            maxVal = Math.max(maxVal, cnt);
        }

        System.out.println(maxVal);
    }
}